From 7cb12d5574ef429b3ca267c5bc3658280859fd92 Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal.vasilek@nic.cz>
Date: Thu, 5 May 2022 15:13:43 +0200
Subject: [PATCH] syncthing: add a patch to increase UDP buffer size

syncthing uses quic-go which warns about insufficient buffer size, so let's increase it:
failed to sufficiently increase receive buffer size (was: 176 kiB, wanted: 2048 kiB, got: 352 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
Also reference: https://github.com/openwrt/packages/pull/18452

Signed-off-by: Michal Vasilek <michal.vasilek@nic.cz>
---
 .../syncthing/files/etc/sysctl.d/90-syncthing-udp-buffer.conf  | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 utils/syncthing/files/etc/sysctl.d/90-syncthing-udp-buffer.conf

diff --git a/utils/syncthing/files/etc/sysctl.d/90-syncthing-udp-buffer.conf b/utils/syncthing/files/etc/sysctl.d/90-syncthing-udp-buffer.conf
new file mode 100644
index 000000000..334dabb4d
--- /dev/null
+++ b/utils/syncthing/files/etc/sysctl.d/90-syncthing-udp-buffer.conf
@@ -0,0 +1,3 @@
+# syncthing uses quic-go which warns about insufficient buffer size, so let's increase it
+# failed to sufficiently increase receive buffer size (was: 176 kiB, wanted: 2048 kiB, got: 352 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
+net.core.rmem_max=2500000
-- 
2.36.0

